### ISSUE TITLE
Can we have CSS bundle lazy-loading when page hydrates?

### ISSUE THREAD

**Issue by Serpentarius13** [shishkin/astro-pagefind#92]
I have a search button with modal popup that renders search bar, so it is very unnecessary to include multple dozens of css into my initial chunk. I think it should be fairly easy to add ability to lazy load css import. I can submit a PR

**Comment by Serpentarius13** [comment]
I'm unable to prevent the component from loading with anything.. I even tried resolving aliases in Vite differently and constructing the component myself.. Client directives do not work on astro component and passing it through children doesn't work because it still loads everything

**Comment by mseele** [comment]
Just found out that loading a stylesheet dynamically is not possible with astro: https://github.com/withastro/astro/issues/9495#issuecomment-2108523677

I'm now doing it with [global styles](https://docs.astro.build/en/guides/styling/#global-styles).

**Comment by Serpentarius13** [comment]
> Just found out that loading a stylesheet dynamically is not possible with astro: [withastro/astro#9495 (comment)](https://github.com/withastro/astro/issues/9495#issuecomment-2108523677)
> 
> I'm now doing it with [global styles](https://docs.astro.build/en/guides/styling/#global-styles).

Yeah, well, the thing is that the library incapsulates import itself and although it is possible to override it, it isn't worth it. I think there should be a way to render and load this component dynamically without leaving the library because it optimizes vite build process nicely. Also, I have gotten this message now:
![image](https://github.com/user-attachments/assets/40c4cd8e-1321-4058-84b1-cccf33178be0)
Although I don't have it on my screen yet, so that means autofocus is also a no-go

**Comment by shishkin** [comment]
I'll close this issue for now. Let me know if there is still anything that can be done in this extension.
